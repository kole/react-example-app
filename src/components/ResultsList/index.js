import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Question from '~/components/Question';
import VisualResult from '~/components/VisualResult';

const Results = styled.div`
    > ul li {
        align-items: center;
        border-bottom: 1px solid #ccc;
        display: flex;
        font-size: 1.2em;
        justify-content: flex-start;
        padding: 10px 0;
        text-align: left;

        i {
            padding-right: 20px;
        }
    }
`;

const ResultsListComponent = (props) => {
    const { questions, answers } = props;

    return (
        <Results>
            <ul>
                {
                    questions.map((questionObj, i) => {
                        const givenAnswer = answers[i].answer;
                        return (
                            <li key={`${i}-question`}>
                                <VisualResult
                                    correctAnswer={questionObj.correct_answer}
                                    givenAnswer={givenAnswer}
                                />
                                <Question questionText={questionObj.question} />
                            </li>
                        );
                    })
                }
            </ul>
        </Results>
    );
};

ResultsListComponent.propTypes = {
    questions: PropTypes.any.isRequired,
    answers: PropTypes.any.isRequired
};

export default ResultsListComponent;
