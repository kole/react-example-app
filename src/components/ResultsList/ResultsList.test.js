/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import ResultsList from './index.js';

const mocks = {
    answers: [{id: 1, answer: true}, {id: 1, answer: true}],
    questions: [
        {
            question: "Sample correct question for test",
            correct_answer: "True"
        },
        {
            question: "Sample incorrect question for test",
            correct_answer: "False"
        }
    ]
};

describe('ResultsList', () => {
    test('Snapshot for correct answer', () => {
        const component = renderer
            .create(
                <ResultsList
                    answers={mocks.answers}
                    questions={mocks.questions}
                />
            )
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
