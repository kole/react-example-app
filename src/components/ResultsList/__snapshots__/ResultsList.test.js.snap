// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList Snapshot for correct answer 1`] = `
<div
  className="sc-htpNat hRzFNt"
>
  <ul>
    <li>
      <div
        className="sc-bwzfXH GTHTj"
      >
        <i
          className="material-icons right"
        >
          check
        </i>
      </div>
      <div
        className="sc-bdVaJa UhWAf"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Sample correct question for test",
          }
        }
      />
    </li>
    <li>
      <div
        className="sc-bwzfXH GTHTj"
      >
        <i
          className="material-icons wrong"
        >
          close
        </i>
      </div>
      <div
        className="sc-bdVaJa UhWAf"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Sample incorrect question for test",
          }
        }
      />
    </li>
  </ul>
</div>
`;
